File: web/src/app/shared/interfaces/module-stats.ts
import { PaymentType } from '../enums/payment-type';
import { ModuleType } from '../enums/module-type';
import { SubscriptionModel } from '@shared/interfaces/subscription-model';

export interface ModuleStats extends SubscriptionModel{
  paymentType: PaymentType;
  type: ModuleType;
}

File: api/src/main/java/se/bjornlunden/blappadminapi/dto/blapp/ModuleStats.java
package se.bjornlunden.blappadminapi.dto.blapp;

import lombok.Builder;
import lombok.Data;
import se.bjornlunden.blappadminapi.enums.ModuleType;
import se.bjornlunden.blappadminapi.enums.PaymentType;

/**
 * @author Patrik Holmkvist on 2021-05-18
 */
@Data
@Builder
public class ModuleStats {
    private Long id;
    private String name;
    private Double price;
    private PaymentType paymentType;
    private ModuleType type;
    private Long subscribers;
    private Long allInclusiveSubscribers;
    private Double totalCost;
}
